metaflow/client/core.py

```diff
@@ -531,12 +533,16 @@ def __getstate__(self):
         from this object) are pickled (serialized) in a later version of Metaflow, it may not be possible
         to unpickle (deserialize) them in a previous version of Metaflow.
         """
+        # Note that we set _namespace_check to False because we want the user to
+        # be able to access this object even after unpickling it. If we set it to
+        # True, it would check the namespace again at the time of unpickling even
+        # if the user properly got the object in the first place and pickled it.
         return {
             "version": "2.8.4",
```

Comment:
this will still be a problem with older versions that do not understand this new dictionary format. how do we handle that case? One option is to not use this custom dictionary format using a version